src/snowflake/cli/_plugins/nativeapp/entities/application.py

```diff
@@ -470,6 +557,15 @@ def create_or_upgrade_app(
                         )
                         print_messages(console, upgrade_cursor)
 
+                        if authorize_event_sharing is not None:
+                            sql_executor.execute_query(
+                                f"alter application {app_name} set AUTHORIZE_TELEMETRY_EVENT_SHARING = {str(authorize_event_sharing).upper()}"
+                            )
```

Comment:
can we put these new queries in the facade? it'll make writing tests easier since you can just mock the function call

Reply:
I tried this out, but this complicates tests as we have to deal with 2 different cursors in all of the tests - I would rather separate this effort from this PR if possible.